src/Clapi/Types/SequenceOps.hs

```diff
@@ -0,0 +1,79 @@
+{-# OPTIONS_GHC -Wall -Wno-orphans #-}
+
+module Clapi.Types.SequenceOps
+  ( ReorderBundle(..), SequenceOp(..)
+  , digest, produce, applyDigest
+  ) where
+
+import Prelude hiding (fail)
+import Control.Monad.Fail (MonadFail(..))
+import qualified Data.Map as Map
+import qualified Data.List as List
+import Data.Foldable (foldlM)
+
+import Clapi.Types.UniqList (UniqList, unUniqList, mkUniqList)
+
+data SequenceOp i
+  = AddAfter (Maybe i)
+  | MoveAfter (Maybe i)
+  | DelElem
+  deriving (Eq, Show)
+
+newtype ReorderBundle i = ReorderBundle [(i, SequenceOp i)] deriving (Eq, Show)
```

Comment:
Out of interest, why did you pick a whole new bundle/digest type, rather than making reordering replace the `ChildAssignments` stuff in the existing bundles?
Are we going to do the `add` and `remove` operations on paths? Like: "Add /engine/loaded_elems/reverb" to load an element without having to care about ordering, and visa versa for deleting it?

Reply:
Why didn't I change the bundles?
 - Because this spins off master and so the types are pretty different.
 - The reordering thing felt independent to any specific ordered UniqList we'd be manipulating.
As for whether we want `add`/`remove` ops on paths, I don't know you end up in 2 ways to do a thing land, but it translates pretty directly. Will have a think about it.

Reply:
If we want to have an "I don't care about ordering" thing, we could change the `Maybe` out for another wrapper type that has "beginning", "end", and "don't care" options rather than having a totally different operation?